ui/src/preview/preview-view.tsx

```diff
@@ -203,16 +206,31 @@ export const PreviewView: React.FC<PreviewProps> = (props: PreviewProps) => {
 
             const validationId = latestValidationIdRef.current + 1;
             latestValidationIdRef.current = validationId;
-            setShowValidationStatusSpinner(true);
+            setPublicationValidationState('IN_PROGRESS');
 
             return validateDebounced(
                 props.layoutContext.branch,
                 props.stagedPublicationCandidateReferences,
-            ).finally(() => {
-                if (latestValidationIdRef.current === validationId) {
-                    setShowValidationStatusSpinner(false);
-                }
-            });
+            )
+                .then((result) => {
+                    if (latestValidationIdRef.current === validationId) {
+                        setPublicationValidationState('API_CALL_OK');
+                    }
+
+                    return result;
+                })
+                .catch((_err) => {
+                    setPublicationValidationState('API_CALL_ERROR');
+
+                    return Promise.resolve({
+                        validatedAsPublicationUnit: stagedPublicationCandidates.map(
+                            setValidationStateToApiError,
+                        ),
+                        allChangesValidated: publicationCandidates.map(
+                            setValidationStateToApiError,
+                        ),
```

Comment:
Hieman outo rakenne: Tässä on ns. "globaali koko esikatselunäkymän tila", ja jokaiselle rivientrylle erikseen omansa... jotka taas ovat ainoastaan ylemmässä taulukossa käytössä. Pelkästään esikatselu-globaalia tilaa voisi toki myös käyttää, mutta silloin molempien taulukoiden jokaiselle rivientrylle näytettäisiin esim. spinneri samaan aikaan. Nythän siis ylemmän taulukon virheitä voi esimerkiksi lueskella normaalisti, vaikka alemmassa taulukossa spinneri pyörii (kaikilla riveillä). Näin siis käy, vaikka ylemmän taulukon tila voi kyllä päivittyä samaan aikaan alemman kanssa, koska kummankin taulukon validaation rajapintakutsut tehdään yhdessä HTTP-kutsussa.